universe/System.cpp

```diff
@@ -310,6 +310,32 @@ bool System::Contains(int object_id) const {
     return m_objects.find(object_id) != m_objects.end();
 }
 
+bool System::ContainsHostileFleet(int empire_id) const {
+    for (auto obj_id : m_objects) {
+        auto object = GetUniverseObject(obj_id);
```

Comment:
Change this to 
````
auto fleet = GetFleet(id)
if (!fleet)
    continue;
````
and remove the next 12 lines.
The intent is clearer and the code is shorter.